### ISSUE TITLE
Why is training so slow

### ISSUE THREAD

**Issue by lilslsls** [yjh0410/yolov2-yolov3_PyTorch#34]
4900 images and batchsize：64  gpu：2080TI

**Comment by yjh0410** [comment]
Because the workers in dataloader is 0. I denote this in my train code. If I add workers, it will report some errors when I use multiscale training trick. I have no idea how to solve this problems.&nbsp;



---Original---

**Comment by yjh0410** [comment]
And if you don't plan to try multi scale training, you'd better set num_workers as 8 or more to accelerate training.



---Original---

**Comment by lilslsls** [comment]
I used yolov2 training, not multi-scale, and a batch spent nearly one minute in training. An epoch has many batches. So I feel strange. Is it because the code parses XML.

**Comment by yjh0410** [comment]
No multi scale？Aha！You can set num_workers as 8(default=8).&nbsp;


Do you set --cuda to use GPU？ Its default value is False.



---Original---

**Comment by lilslsls** [comment]
Sorry, I didn't read the code carefully. Now it's fast to use CUDA. thank you

**Comment by yjh0410** [comment]
You're welcome！！！(๑&gt;؂<๑）



---Original---

**Comment by bruce1408** [comment]
> No multi scale？Aha！You can set num_workers as 8(default=8).&nbsp; Do you set --cuda to use GPU？ Its default value is False.
> […](#)
> ---Original--- From: "lilslsls"<notifications@github.com&gt; Date: Sat, May 23, 2020 14:14 PM To: "yjh0410/yolov2-yolov3_PyTorch"<yolov2-yolov3_PyTorch@noreply.github.com&gt;; Cc: "Comment"<comment@noreply.github.com&gt;;"Jianhua Yang"<1394571815@qq.com&gt;; Subject: Re: [yjh0410/yolov2-yolov3_PyTorch] Why is training so slow (#34) I used yolov2 training, not multi-scale, and a batch spent nearly one minute in training. An epoch has many batches. So I feel strange. Is it because the code parses XML. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**SET cuda TRUE**

- 2080ti x 1
- ubuntu 16.04
- python 3.6
- pytorch 1.5.0
- cpu cores  :8
- cpu processor num :16

when i use cpu to train the model ,the training is sooooooo slow, such as the blow info...
[Epoch 1/250][Iter 0/517][lr 0.000000][Loss: obj 210.70 || cls 4.88 || bbox 8.88 || total 224.46 || size 416 || time: 47.35]

but when i set cuda True, the training speed up  obivous! 
Epoch 1/250][Iter 10/517][lr 0.000000][Loss: obj 212.88 || cls 7.12 || bbox 9.07 || total 229.07 || size 416 || time: 2.20]

**Comment by bruce1408** [comment]
**SET CUDA TRUE To use GPU!**
### envs
- 2080ti x 1
- ubuntu 16.04
- python 3.6
- pytorch 1.5.0
- cpu cores  :8
- cpu processor num :16
###  without GPU
when i use cpu to train the model ,the training is sooooooo slow, such as the blow info...
[Epoch 1/250][Iter 0/517][lr 0.000000][Loss: obj 210.70 || cls 4.88 || bbox 8.88 || total 224.46 || size 416 || time: 47.35]

### use GPU
but when i set cuda True, the training speed up  obivous! 
Epoch 1/250][Iter 10/517][lr 0.000000][Loss: obj 212.88 || cls 7.12 || bbox 9.07 || total 229.07 || size 416 || time: 2.20]

**Comment by yjh0410** [comment]
Thanks for your advice！HAHA！

The reason why I didn't set cuda as True is that I wanna indirectly remind everyone who plans to try this project to use GPU to train the model. If no GPU，it's a little hard to study object detection.

**Comment by yjh0410** [comment]
> **SET CUDA TRUE To use GPU!**
> 
> ### envs
> * 2080ti x 1
> * ubuntu 16.04
> * python 3.6
> * pytorch 1.5.0
> * cpu cores  :8
> * cpu processor num :16
> 
> ### without GPU
> when i use cpu to train the model ,the training is sooooooo slow, such as the blow info...
> [Epoch 1/250][Iter 0/517][lr 0.000000][Loss: obj 210.70 || cls 4.88 || bbox 8.88 || total 224.46 || size 416 || time: 47.35]
> 
> ### use GPU
> but when i set cuda True, the training speed up obivous!
> Epoch 1/250][Iter 10/517][lr 0.000000][Loss: obj 212.88 || cls 7.12 || bbox 9.07 || total 229.07 || size 416 || time: 2.20]

Thanks for your advice！HAHA！

The reason why I didn't set cuda as True is that I wanna indirectly remind everyone who plans to try this project to use GPU to train the model. If no GPU，it's a little hard to study object detection.

**Comment by bruce1408** [comment]
> > **SET CUDA TRUE To use GPU!**
> > ### envs
> > 
> > * 2080ti x 1
> > * ubuntu 16.04
> > * python 3.6
> > * pytorch 1.5.0
> > * cpu cores  :8
> > * cpu processor num :16
> > 
> > ### without GPU
> > when i use cpu to train the model ,the training is sooooooo slow, such as the blow info...
> > [Epoch 1/250][Iter 0/517][lr 0.000000][Loss: obj 210.70 || cls 4.88 || bbox 8.88 || total 224.46 || size 416 || time: 47.35]
> > ### use GPU
> > but when i set cuda True, the training speed up obivous!
> > Epoch 1/250][Iter 10/517][lr 0.000000][Loss: obj 212.88 || cls 7.12 || bbox 9.07 || total 229.07 || size 416 || time: 2.20]
> 
> Thanks for your advice！HAHA！
> 
> The reason why I didn't set cuda as True is that I wanna indirectly remind everyone who plans to try this project to use GPU to train the model. If no GPU，it's a little hard to study object detection.

**Your purpose is good, but I still hope the cuda default setting is True, thx!!**
